repositories/profile.ts

```diff
@@ -0,0 +1,44 @@
+import { Schema, Document } from 'mongoose';
+import ProfileModel from '../dbmodels/profile';
+import HTTPError from '../exceptions/HTTPError';
+import IProfile from '../types/dbtypes/profile';
+
+// Gets a profile by id
+export const get_profile_by_id = async (id: string): Promise<IProfile> => {
+  return await ProfileModel.findById(id).then(
```

Comment:
This only handles errors returned because a profile wasn't found. Instead of this closure, we should simplify this to:
```
try {
  const docs: IProfile | null | undefined = await ProfileModel.findById(id);
  if (!docs) {
     return Promise.reject(new HTTPError('Profile not found', 404));
  }
} catch (e) {
    // this should be mongo specific errors, so perhaps we return some error on the server side....?
    return Promise.reject(new HTTPError('Ahhh an unknown error?', 500));
}